libvirt.go

```diff
@@ -618,16 +615,32 @@ func getQEMUError(r response) error {
 // New configures a new Libvirt RPC connection.
 func New(conn net.Conn) *Libvirt {
 	l := &Libvirt{
-		conn:      conn,
-		s:         0,
-		r:         bufio.NewReader(conn),
-		w:         bufio.NewWriter(conn),
-		mu:        &sync.Mutex{},
-		callbacks: make(map[int32]chan response),
-		events:    make(map[int32]*event.Stream),
+		conn:              conn,
+		s:                 0,
+		r:                 bufio.NewReader(conn),
+		w:                 bufio.NewWriter(conn),
+		mu:                &sync.Mutex{},
+		socketCleanupDone: make(chan struct{}),
+		callbacks:         make(map[int32]chan response),
+		events:            make(map[int32]*event.Stream),
 	}
 
-	go l.listen()
+	go func() {
```

Comment:
not a big deal, I'd prefer if this were in a teardown-centric fn on `Libvirt`, making this read more like:

```go
go l.waitForShutdown()
```

Reply:
yeah... I definitely prefer listenAndWait()

Reply:
ended up going with listenAndRoute